lib/kitchen/verifier/pester.rb

```diff
@@ -148,148 +199,173 @@ def run_command_script
+            <<-PS1
+              ${#{powershell_module[:Name]}} = #{ps_hash(powershell_module)}
+              Install-ModuleFromNuget -Module ${#{powershell_module[:Name]}} #{gallery_url_param}
+            PS1
+          else
+            <<-PS1
+              Install-ModuleFromNuget -Module @{Name = '#{powershell_module}'} #{gallery_url_param}
+            PS1
+          end
+        end
+      end
+
+      # Returns the string command to set a PS Repository
+      # for each PSRepo configured.
+      #
+      # @return [Array<String>] array of suite files
+      # @api private
+      def register_psrepository
+        return if config[:register_repository].nil?
+
+        info("Registering a new PowerShellGet Repository")
+        Array(config[:register_repository]).map do |psrepo|
+          # Using Set-PSRepo from ../../*/*/*/PesterUtil.psm1
+          debug("Command to set PSRepo #{psrepo[:Name]}.")
+          <<-PS1
+            Write-Host 'Registering psrepo #{psrepo[:Name]}...'
+            ${#{psrepo[:Name]}} = #{ps_hash(psrepo)}
+            Set-PSRepo -Repository ${#{psrepo[:Name]}}
+          PS1
+        end
+      end
+
+      def install_pester
+        return if config[:use_local_pester_module]
+
+        pester_install_params = config[:pester_install] || {}
+        <<-PS1
+          if ((Get-PSRepository -Name PSGallery).InstallationPolicy -ne 'Trusted') {
+              Write-Host -Object "Trusting the PSGallery to install Pester without -Force"
+              Set-PSRepository -Name PSGallery -InstallationPolicy Trusted -ErrorAction SilentlyContinue
+          }
+
+          Write-Host "Installing Pester..."
+          $InstallPesterParams = #{ps_hash(pester_install_params)}
+          $InstallPesterParams['Name'] = 'Pester'
+          Install-module @InstallPesterParams
+          Write-Host 'Pester Installed.'
+        PS1
+      end
+
+      def install_modules_from_gallery
+        return if config[:install_modules].nil?
+
+        Array(config[:install_modules]).map do |powershell_module|
+          if powershell_module.is_a? Hash
+            # Sanitize variable name so that $powershell-yaml becomes $powershell_yaml
+            module_name = powershell_module[:Name].gsub(/[\W]/, "_")
+            # so we can splat that variable to install module
+            <<-PS1
+              $#{module_name} = #{ps_hash(powershell_module)}
+              Write-host -noNewline 'Instaling #{module_name}'
+              Install-Module @#{module_name}
+              Write-host '... done.'
+            PS1
+          else
+            <<-PS1
+              Write-host -noNewline 'Installing #{powershell_module} ...'
+              Install-Module -Name '#{powershell_module}'
+              Write-host '... done.'
+            PS1
+          end
+        end
       end
 
       def really_wrap_shell_code(code)
-        wrap_shell_code(Util.outdent!(use_local_powershell_modules(code)))
+        if windows_os?
```

Comment:
instead of doing an `if` with an indent, use a guard clause.

```
return wrap_shell_code(Util.outdent!(use_local_powershell_modules(code))) if windows_os?
```

or perhaps abstract it further: e.g.

```
def really_wrap_shell_code(code)
  windows_os? ? really_wrap_windows_shell_code(code) : really_wrap_posix_shell_code(code)
end

def really_wrap_windows_shell_code(code)
  ...
end

def really_wrap_posix_shell_code(code)
  ...
end
```

Reply:
Hmm. I think it'll be clearer to abstract it one layer.

Reply:
Additional Q: is it worth having a branch here for MacOS, or is it safe enough to treat it similarly to Linux?

Reply:
I think it should be the same for MacOS, it's mainly about shebang, sudo and wrapping for pwsh invocation...